core/lib/web3_decl/src/error.rs

```diff
@@ -42,3 +55,152 @@ pub enum Web3Error {
     #[error("Tree API is not available")]
     TreeApiUnavailable,
 }
+
+/// RPC error with additional details: the method name and arguments of the called method.
+///
+/// The wrapped error can be accessed using [`AsRef`].
+#[derive(Debug)]
+pub struct EnrichedRpcError {
```

Comment:
This is generally for _clients_ to add additional information about requests made, right?
I was initially a bit confused assuming that it can be used on the server side too, and it doesn't look like this.

If so, I wonder whether we make it a bit less confusing, e.g. by changing the naming (`Enrich`_`Client`_`RpcError`) or moving it elsewhere (create some client-specific module? `client_error_ext`?).

Reply:
Yeah, I think using the original `ClientError` name instead of `RpcError` would make things clearer.